wazuh/wazuh-dashboard-plugins
# Added limit to FIM csv export

### Description

This PR adds: 
- A setting to set the maximum number of rows to be included in csv reports
- A warning icon explaining that that will only be visible if there are more results than the rows limit.
- The table's sorting as a filter for exporting csv, to include the results in the reports as they are shown in the table
 
### Issues Resolved

- https://github.com/wazuh/wazuh-dashboard-plugins/issues/7068

### Evidence

![imagen](https://github.com/user-attachments/assets/18841c37-5267-4bf7-83c2-0dbd1f7d6be9)

![Peek 12-12-2024 11-38](https://github.com/user-attachments/assets/14f0f7e7-2170-4c04-abff-4d6f1fb04d48)

**Setting the limit to 1439**
![imagen](https://github.com/user-attachments/assets/dd5997c6-86ac-4839-b059-99667a25f369)


### Test
**Pre requisites**
- Bring up an environment with a real manager and an agent
- Add a high number of files in the agent for the FIM. 
  - By default it reads the directory `/etc`. You can create the following file inside `/etc`
    ```
    for i in {1..12000}; do 
        echo "TEST" >> "file$i.txt" 
      done 
    ```
  And execute it with `bash <filename>`. After that, restart the agent and wait for all the files to be read (it can take some time).
- Navigate to `File Integrity Monitoring`/`Inventory` and select the agent.

**Test**
- Verify that there are more than 10.000 files and the warning message appears.
- Click on `Export formatted` and verify that the exported file only contains 10.000 results.
- Change the sorting in the table, click on `Export formatted` and verify that the exported file is sorted like the table.
- Apply some search term to reduce the number of files to less than 10.000 and verify that the warning is not shown.
- Click on `Export formatted` and verify that the exported report contains the same results shown in the table.

- Navigate to `Dashboard management > App Settings` and change the `reports.csv.maxRows` setting to a different number
- Repeat the previous steps considering the new set value

### Check List
